### PR TITLE
Add MySQL Flexible Server resources

### LINKED ISSUE #1966: Feature: Support MySQLFlexibleServer

### PR DESCRIPTION
**Pull request by babbageclunk**
**What this PR does / why we need it**:
Adds support for MySQL FlexibleServer, FlexibleServersFirewallRule and FlexibleServersDatabase

Closes #1966 

**Special notes for your reviewer**:
Generated code is in a separate commit to make it easier to avoid.

The RP had a couple of quirks, I'm mentioning them to our contacts on that team:
* No ARM ID coming back when getting the FirewallRule
* Internal server error (rather than helpful message) when creating a database with a hyphenated name

**How does this PR make you feel**:
gif

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: v2/api/resources.md [hunk 1]
```diff
@@ -43,6 +43,17 @@ Use CRD version `v1alpha1api20201201`
 
 Use CRD version `v1alpha1api20210501`
 
+### microsoft.dbformysql
+
+
+#### ARM version 2021-05-01
+
+- FlexibleServer ([sample](https://github.com/Azure/azure-service-operator/tree/main/v2/config/samples/microsoft.dbformysql/v1alpha1api20210501_flexibleserver.yaml))
+- FlexibleServersDatabase ([sample](https://github.com/Azure/azure-service-operator/tree/main/v2/config/samples/microsoft.dbformysql/v1alpha1api20210501_flexibleserversdatabase.yaml))
```
Comment (matthchr) [inline v2/api/resources.md:52]: Is there a corresponding `FlexibleServerConfiguration` which we also added support for in Postgres?

**Review by matthchr — APPROVED** [review]
LGTM. Had one minor comment. If that resource exists, feel free to do it in this PR or merge this PR and do it later (or file an item for it or whatever).

File: v2/api/resources.md (hunk 1 above)
Comment (babbageclunk) [reply v2/api/resources.md:52]: No, I looked for that but there isn't a corresponding one in the MySQL resources:
https://docs.microsoft.com/en-us/azure/templates/microsoft.dbformysql/allversions
vs
https://docs.microsoft.com/en-us/azure/templates/microsoft.dbforpostgresql/allversions

**Pull request merged**